test/position-managers/PositionManager.modifyLiquidities.t.sol

```diff
@@ -696,4 +707,130 @@ contract PositionManagerModifyLiquiditiesTest is Test, PosmTestSetup, LiquidityF
         vm.expectRevert(DeltaResolver.InsufficientBalance.selector);
         lpm.modifyLiquidities(actions, _deadline);
     }
+
+    function test_mintFromDeltas_fot() public {
+        // Use a 1% fee.
+        MockFOT(address(fotToken)).setFee(100);
+        uint256 tokenId = lpm.nextTokenId();
+
+        uint256 fotBalanceBefore = Currency.wrap(address(fotToken)).balanceOf(address(this));
+
+        uint256 amountAfterTransfer = 990e18;
+        uint256 amountToSendFot = 1000e18;
+
+        (uint256 amount0, uint256 amount1) = fotKey.currency0 == Currency.wrap(address(fotToken))
+            ? (amountToSendFot, amountAfterTransfer)
+            : (amountAfterTransfer, amountToSendFot);
+
+        // Calculcate the expected liquidity from the amounts after the transfer. They are the same for both currencies.
+        uint256 expectedLiquidity = LiquidityAmounts.getLiquidityForAmounts(
+            SQRT_PRICE_1_1,
+            TickMath.getSqrtPriceAtTick(LIQUIDITY_PARAMS.tickLower),
+            TickMath.getSqrtPriceAtTick(LIQUIDITY_PARAMS.tickUpper),
+            amountAfterTransfer,
+            amountAfterTransfer
+        );
+
+        Plan memory planner = Planner.init();
+        planner.add(Actions.SETTLE, abi.encode(fotKey.currency0, amount0, true));
+        planner.add(Actions.SETTLE, abi.encode(fotKey.currency1, amount1, true));
+        planner.add(
+            Actions.MINT_POSITION_FROM_DELTAS,
```

Comment:
I’m surprised you don’t need a take at the end for any excess of each token that was left in the contract if one of them was too big?

Reply:
Its because I calculate the amounts exactly pretty much. This won't work for most use cases where you don't know the exact amounts in both token0 and token1